plugins/statistics.koplugin/main.lua

```diff
@@ -306,7 +308,21 @@ Do you want to create an empty database?
             logger.info("ReaderStatistics: Migrating DB from schema", db_version, "to schema", DB_SCHEMA_VERSION, "...")
             -- Backup the existing DB first
             conn:close()
-            local bkp_db_location = db_location .. ".bkp." .. db_version .. "-to-" .. DB_SCHEMA_VERSION
+            local bkp_base_location = db_location .. ".bkp." .. db_version .. "-to-" .. DB_SCHEMA_VERSION
+            local bkp_db_location = bkp_base_location
+            -- Don't overwrite an existing backup
+            local bkp_rev = 1
+            while C.access(bkp_db_location, C.F_OK) == 0 do
```

Comment:
What's wrong with our usual `lfs.attributes(file, "mode") == "file"` ? :)

I'd say that if there is one that exists, and if we're re-migrating (so, because something failed and the current db is bad), just don't do anything - no need to keep backups of bad dbs. ?

Reply:
> What's wrong with our usual `lfs.attributes(file, "mode") == "file"` ? :)
> 

I wasn't super-convinced it wouldn't do weird things in a while loop, so I stuck with what I knew would work ;p.

> I'd say that if there is one that exists, and if we're re-migrating (so, because something failed and the current db is bad), just don't do anything - no need to keep backups of bad dbs. ?

It's indeed an *extremely* defensive approach, true.
Then again, it's still useful to debug *what* exactly failed to migrate ;).

Reply:
> > What's wrong with our usual `lfs.attributes(file, "mode") == "file"` ? :)
> 
> I wasn't super-convinced it wouldn't do weird things in a while loop, so I stuck with what I knew would work ;p.
> 

Okay, it's essentially just a wrapper around `stat`, so, that works ;). A tiny bit overkill, but safe.

Reply:
> > I'd say that if there is one that exists, and if we're re-migrating (so, because something failed and the current db is bad), just don't do anything - no need to keep backups of bad dbs. ?
> 
> It's indeed an _extremely_ defensive approach, true.
> Then again, it's still useful to debug _what_ exactly failed to migrate ;).

Well, there's the actual botched migrated db for that, duh'. >_<".

Reply:
(Yes, I'm talking to myself, don't mind me, I haven't slept much ^^)